packages/hydrogen/src/getStorefrontHeaders.ts

```diff
@@ -0,0 +1,24 @@
+import type {CrossRuntimeRequest} from './utils/request';
+
+export type StorefrontHeaders = {
+  /** A unique ID that correlates all sub-requests together. */
+  requestGroupId: string | null;
+  /** The IP address of the client. */
+  buyerIp: string | null;
+  /** The cookie header from the client  */
+  cookie: string | null;
+  /** The purpose header value for debugging */
+  purpose: string | null;
+};
+
+export function getStorefrontHeaders(
+  request: Request | CrossRuntimeRequest,
+): StorefrontHeaders {
+  const headers = request.headers;
+  return {
+    requestGroupId: (headers.get ? headers.get('request-id') : null) || null,
+    buyerIp: (headers.get ? headers.get('oxygen-buyer-ip') : null) || null,
+    cookie: (headers.get ? headers.get('cookie') : null) || null,
+    purpose: (headers.get ? headers.get('purpose') : null) || null,
+  };
+}
```

Comment:
The idea of having `getStorefrontHeaders` in `remix-oxygen` package is because it's unique to the Oxygen platform. In Node or somewhere else, you'd get buyerIp, cookie, etc. in other ways.

Therefore, if we want to move this function here, at the very least I think we should make it obvious this only works on Oxygen or it will be misleading. Perhaps `getStorefrontFromOxygenHeaders`?

Reply:
The main reason why I was moving getStorefrontHeaders into @Shopify/hydrogen was because I can't import @shopify/remix-oxygen. It created circular import and won't build.

Maybe what I can do is just a local method that account for both without exporting it at all?

Reply:
Looking through the storefront code, looks like we generate a UUID for `requestGroupId`, `buyerIp` and `cookie` gets default to empty string, and `purpose` us just left as null as default.

I dont think we should throw since we currently dont, maybe warn? Not sure if we really need to thou.

Reply:
>Looking through the storefront code, looks like we generate a UUID for requestGroupId, buyerIp and cookie gets default to empty string, and purpose us just left as null as default.
>I dont think we should throw since we currently dont, maybe warn? Not sure if we really need to thou.

True, I thought these params were required in TS at least, but they are actually optional. In that case we probably don't need to throw/warn, although we should probably consider warning eventually...